### PR TITLE
LPD-43468 Trim all titles to have the max size of name column

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-43468 When creating/editing a Category if you ignore the Front validator, you can add title on non-default languages on more than 255 characters

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-43468

When you are creating or editing a Category a validation of the title is shown, but you can skip this validation if you change the language. So this can allow users to add a title for a language that is bigger that 255 characters, that is the max value of the name column and in the case of the default language, is trimmed to that value on backend

# How am I fixing it?

Trim all the titles, are the default or not. 

# How can you verify that it works?

Run the included automated tests.

# Commits

- **LPD-43468 use service instead util**
- **LPD-43468 add test to check that all the titles are trimmed**
- **LPD-43468 trim all the titles, not only the default locale**

### CODE REVIEW

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryLocalServiceTest.java [hunk 1]
```diff
@@ -430,6 +429,54 @@ public void testCategoryWithLongNameIsTrimmed() throws Exception {
 		Assert.assertEquals(assetCategoryName, assetCategory.getName());
 	}
 
+	@Test
+	public void testCategoryWithLongTitlesAreTrimmed() throws Exception {
```
Comment (robertoDiaz) [inline modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryLocalServiceTest.java:433]: Use AssetCategory in title pattern

**Review by robertoDiaz — CHANGES_REQUESTED** [review]
Just the test title 

Good work!

File: modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryLocalServiceTest.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryLocalServiceTest.java:433]: I'm following the pattern of testCategoryWithLongNameIsTrimmed.

in your opinion, what do you recommend?:
- change both to "assetCategory"
- change testCategoryWithLongNameIsTrimmed to testAddCategoryWithLongNameIsTrimmed (method name) and then split the new tests on testAddCategoryWithLongTitlesAreTrimmed and testUpdateCategoryWithLongTitlesAreTrimmed
- only change the new method.


Thanks for your input

Comment (robertoDiaz) [reply modules/apps/asset/asset-categories-test/src/testIntegration/java/com/liferay/asset/categories/service/test/AssetCategoryLocalServiceTest.java:433]: mmm, you are right.

Let's keep it as it is.

**Review by robertoDiaz — APPROVED** [review]

**Comment by robertoDiaz** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:stop

**Comment by adolfopa** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
[com.liferay.asset.categories.service.test.AssetCategoryLocalServiceTest](https://test-1-38.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/992500/testReport/com.liferay.asset.categories.service.test/AssetCategoryLocalServiceTest) - [Failed 0 of last 18](https://testray.liferay.com/home/-/testray/case_results/86942582/history) was relevant to this PR

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:report:87890259

**Comment by AliciaGarciaGarcia** [comment]
failures not related with PR

**Comment by AliciaGarciaGarcia** [comment]
ci:forward:force

**Pull request closed without merging**